sycl/source/detail/kernel_impl.hpp

```diff
@@ -251,6 +255,17 @@ class kernel_impl {
   std::mutex *MCacheMutex = nullptr;
   mutable std::string MName;
 
+  using DeviceKernelInfoDeleterTy = void (*)(DeviceKernelInfo *);
+  using DeviceKernelInfoUniquePtr =
+      std::unique_ptr<DeviceKernelInfo, DeviceKernelInfoDeleterTy>;
+  DeviceKernelInfoUniquePtr MDeviceKernelInfo;
+
+  static void DeviceKernelInfoDeleter(DeviceKernelInfo *Ptr) { delete Ptr; }
```

Comment:
is this needed? std::unique_ptr will use `std::default_delete<T>`  and isn't that just plain `delete` ?